hyper/contrib.py

```diff
@@ -43,28 +48,40 @@ def get_connection(self, host, port, scheme, cert=None):
         if cert is not None:
             ssl_context = init_context(cert=cert)
 
+        connection_key = (host, port, scheme, cert, proxy)
```

Comment:
Well, OK, will do, but still, I think storing proxy headers in this key is redundant, because they are generated from a proxy which is already a part of this key